### ISSUE TITLE
Adding item to large array takes very long

### ISSUE THREAD

**Issue by dev-kanishk** [ngx-formly/ngx-formly#3791]
**Description**
When the length of array increases, Add operation on array takes quite a long. In my case it is taking 4 sec.

**Minimal Reproduction**
Create an array which item are nested at-least 2 level and with at least 9 to 10 fields. Now start adding items to array, will observe the amount of time it take to perform add operation increases. 

Formly version: 6.1.8

Additional context:

I found that building array fields is taking a lot of time. Reason is for all the array items and subfields **"extension.postPopulate"** is called, which is further calling the **"setValidators"** method of **"field-form.ts"** here the creation of observable for async validations takes a lot of time.

```
c.setAsyncValidators(() => {
            const v = Validators.composeAsync(this.mergeValidators<AsyncValidatorFn>(field, '_asyncValidators'));
            return v ? v(c) : of(null);
    });
``` 

I did the following POC which reduced this time from **4sec to 0.6sec**.

**What I did**
I added an if condition before the "c.setAsyncValidators" method to check if we have async validators.
```
const asyncValidatorsList =
    this.mergeValidators<AsyncValidatorFn>(field, '_asyncValidators');

asyncValidatorsList.length && c.setAsyncValidators(() => {
     const v = Validators.composeAsync(asynValidatorsList);
     return v ? v(c) : of(null);
   });
```

**Comment by aitboudad** [comment]
Hi, could you please send a PR for the fix, also it would great if you possible to provide a reproduction using `stackblitz`.

**Comment by dev-kanishk** [comment]
Reproduction:
stackblitz:
https://stackblitz.com/edit/angular-kuxmzz-gu3w5u?file=src%2Fapp%2Farray.type.ts,src%2Fapp%2Fapp.component.ts

As soon as the size of array increases time it takes to add new item increases.

![Screenshot 2023-10-06 at 11 45 23 AM](https://github.com/ngx-formly/ngx-formly/assets/48204976/761a9146-17e2-403c-b79a-bdd890da15f3)

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.2.1](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.2.1) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
